python/pyspark/cloudpickle.pyi

```diff
@@ -0,0 +1,86 @@
+#
```

Comment:
I believe this file can be removed. It's a copy of [cloudpickle](https://github.com/cloudpipe/cloudpickle), which should be separate from PySpark. I believe it makes less sense to type this file.

BTW, should we maybe start it with typing only exposed APIs for now? I think having type hints in internal packages has less benefits (only to PySpark dev) and it's arguable to maintain it.

Reply:
> I believe this file can be removed

Indeed, thanks!

> BTW, should we maybe start it with typing only exposed APIs for now? I think having type hints in internal packages has less benefits (only to PySpark dev) and it's arguable to maintain it.

I agree.

In general we might be able to shed some weight here by removing stubs which have no static annotations.  However I'd like to avoid to situation where we'll have to `type: ignore` or exclude to many things.

Reply:
I remember mypy just ignores and doesn't require `type: ignore` when the type hints don't exist. That's what I experienced in another project that uses type hints but maybe I am missing some cases here.

In any event, let's port first in whichever way best to you, and then we can try to remove non-API type hints separately as a followup. Please go ahead in whichever way easiest to you as the initial port!